extensions/amp-apester-media/0.1/amp-apester-media.js

```diff
@@ -101,31 +128,62 @@ class AmpApesterMedia extends AMP.BaseElement {
     this.height_ = getLengthNumeral(height);
     this.random_ = false;
     this.mediaAttribute_ = user().assert(
-        (this.element.getAttribute('data-apester-media-id') ||
-         (this.random_ =
-             this.element.getAttribute('data-apester-channel-token'))),
+      (this.element.getAttribute('data-apester-media-id') ||
+        (this.random_ =
+          this.element.getAttribute('data-apester-channel-token'))),
         'Either the data-apester-media-id or the data-apester-channel-token ' +
-        'attributes must be specified for <amp-apester-media> %s',
+      'attributes must be specified for <amp-apester-media> %s',
         this.element);
+    this.embedOptions_ = {
+      playlist: this.random_,
+      idOrToken: this.mediaAttribute_,
+      inative: this.element.getAttribute('data-apester-inative') === 'true',
+      fallback: this.element.getAttribute('data-apester-fallback'),
+      distributionChannelId:
+        this.element.getAttribute('data-apester-channel-id'),
+      renderer: true,
+      tags: extractTags(this.element, document),
+    };
   }
 
   /** @override */
   firstLayoutCompleted() {
     this.viewportCallback(this.isInViewport());
-
     // Do not hide placeholder
   }
 
   /**
    * @return {string}
    **/
   buildUrl_() {
+    const {
+      idOrToken, playlist, inative, distributionChannelId, fallback,
+      renderer, tags,
+    } = this.embedOptions_;
     const encodedMediaAttribute = encodeURIComponent(
         dev().assertString(this.mediaAttribute_));
-    const suffix = (this.random_) ?
-        `/tokens/${encodedMediaAttribute}/interactions/random` :
-        `/interactions/${encodedMediaAttribute}/display`;
-    return `${this.displayBaseUrl_}${suffix}`;
+    let suffix = '';
+    let queryParmas = '?';
```

Comment:
please turn this into a map to store key/values and then use `addParamToUrl` util from `url.js` to append to the url. the util will handle encoding as well.